### ISSUE TITLE
[Bug]: TPCH Q21 Q22 result are not correct on EKS with 2*cn

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9013]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-5ad2c34
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2023-04-16 17:23:06 Now start to execute the query queries/q21.sql,please wait.....

2023-04-16 17:23:17 The query_id is: 59c23e12-dc7b-11ed-95e2-ba3381988abc
2023-04-16 17:23:17 Now start to compare query result for [report/TPCH_10/q21.txt,golden/TPCH_10/q21.txt]
2023-04-16 17:23:17 The column[0] value of row[1] does not equal to each other, [Supplier#000099056] and [Supplier#000024297]
2023-04-16 17:23:17 The column[1] value of row[1] does not equal to each other, [235] and [22]
2023-04-16 17:23:17 The column[0] value of row[2] does not equal to each other, [Supplier#000000092] and [Supplier#000014526]
2023-04-16 17:23:17 The column[1] value of row[2] does not equal to each other, [234] and [21]
2023-04-16 17:23:17 The column[0] value of row[3] does not equal to each other, [Supplier#000086115] and [Supplier#000050896]
2023-04-16 17:23:17 The column[1] value of row[3] does not equal to each other, [233] and [21]
2023-04-16 17:23:17 The column[0] value of row[4] does not equal to each other, [Supplier#000097341] and [Supplier#000051263]
2023-04-16 17:23:17 The column[1] value of row[4] does not equal to each other, [233] and [21]
2023-04-16 17:23:17 The column[0] value of row[5] does not equal to each other, [Supplier#000024251] and [Supplier#000021732]
2023-04-16 17:23:17 The column[1] value of row[5] does not equal to each other, [232] and [20]
2023-04-16 17:23:17 The column[0] value of row[6] does not equal to each other, [Supplier#000000722] and [Supplier#000077039]
2023-04-16 17:23:17 The column[1] value of row[6] does not equal to each other, [229] and [20]
2023-04-16 17:23:17 The column[0] value of row[7] does not equal to each other, [Supplier#000010033] and [Supplier#0000[91](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:92)469]
2023-04-16 17:23:17 The column[1] value of row[7] does not equal to each other, [229] and [20]
2023-04-16 17:23:17 The column[0] value of row[8] does not equal to each other, [Supplier#000009000] and [Supplier#000097945]
2023-04-16 17:23:17 The column[1] value of row[8] does not equal to each other, [228] and [20]
2023-04-16 17:23:17 The column[0] value of row[9] does not equal to each other, [Supplier#000097330] and [Supplier#0000057[92](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:93)]
2023-04-16 17:23:17 The column[1] value of row[9] does not equal to each other, [228] and [19]
2023-04-16 17:23:17 The column[0] value of row[10] does not equal to each other, [Supplier#000000210] and [Supplier#000006372]
2023-04-16 17:23:17 The column[1] value of row[10] does not equal to each other, [227] and [19]
2023-04-16 17:23:17 The column[0] value of row[11] does not equal to each other, [Supplier#000059195] and [Supplier#000008576]
2023-04-16 17:23:17 The column[1] value of row[11] does not equal to each other, [227] and [19]
2023-04-16 17:23:17 The column[0] value of row[12] does not equal to each other, [Supplier#000068665] and [Supplier#000023020]
2023-04-16 17:23:17 The column[1] value of row[12] does not equal to each other, [227] and [19]
2023-04-16 17:23:17 The column[0] value of row[13] does not equal to each other, [Supplier#000085642] and [Supplier#000030847]
2023-04-16 17:23:17 The column[1] value of row[13] does not equal to each other, [227] and [19]
2023-04-16 17:23:17 The column[0] value of row[14] does not equal to each other, [Supplier#000091323] and [Supplier#000034574]
2023-04-16 17:23:17 The column[1] value of row[14] does not equal to each other, [227] and [19]
2023-04-16 17:23:17 The column[0] value of row[15] does not equal to each other, [Supplier#000002422] and [Supplier#000039817]
2023-04-16 17:23:17 The column[1] value of row[15] does not equal to each other, [226] and [19]
2023-04-16 17:23:17 The column[0] value of row[16] does not equal to each other, [Supplier#000003198] and [Supplier#000062406]
2023-04-16 17:23:17 The column[1] value of row[16] does not equal to each other, [226] and [19]
2023-04-16 17:23:17 The column[0] value of row[17] does not equal to each other, [Supplier#000069126] and [Supplier#000065063]
2023-04-16 17:23:17 The column[1] value of row[17] does not equal to each other, [226] and [19]
2023-04-16 17:23:17 The column[0] value of row[18] does not equal to each other, [Supplier#000039817] and [Supplier#000067630]
2023-04-16 17:23:17 The column[1] value of row[18] does not equal to each other, [224] and [19]
2023-04-16 17:23:17 The column[0] value of row[19] does not equal to each other, [Supplier#000071379] and [Supplier#000079715]
2023-04-16 17:23:17 The column[1] value of row[19] does not equal to each other, [224] and [19]
2023-04-16 17:23:17 The column[0] value of row[20] does not equal to each other, [Supplier#000071457] and [Supplier#000087745]
2023-04-16 17:23:17 The column[1] value of row[20] does not equal to each other, [224] and [19]
2023-04-16 17:23:17 The column[0] value of row[21] does not equal to each other, [Supplier#0000[93](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:94)317] and [Supplier#000089865]
2023-04-16 17:23:17 The column[1] value of row[21] does not equal to each other, [224] and [19]
2023-04-16 17:23:17 The column[0] value of row[22] does not equal to each other, [Supplier#0000[94](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:95)385] and [Supplier#000000921]
2023-04-16 17:23:17 The column[1] value of row[22] does not equal to each other, [224] and [18]
2023-04-16 17:23:17 The column[0] value of row[23] does not equal to each other, [Supplier#000005792] and [Supplier#000002567]
2023-04-16 17:23:17 The column[1] value of row[23] does not equal to each other, [223] and [18]
2023-04-16 17:23:17 The column[0] value of row[24] does not equal to each other, [Supplier#000068547] and [Supplier#000002966]
2023-04-16 17:23:17 The column[1] value of row[24] does not equal to each other, [223] and [18]
2023-04-16 17:23:17 The column[0] value of row[25] does not equal to each other, [Supplier#000069478] and [Supplier#000003947]
2023-04-16 17:23:17 The column[1] value of row[25] does not equal to each other, [223] and [18]
2023-04-16 17:23:17 The column[0] value of row[26] does not equal to each other, [Supplier#000076257] and [Supplier#000023842]
2023-04-16 17:23:17 The column[1] value of row[26] does not equal to each other, [223] and [18]
2023-04-16 17:23:17 The column[0] value of row[27] does not equal to each other, [Supplier#000082222] and [Supplier#0000246[95](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:96)]
2023-04-16 17:23:17 The column[1] value of row[27] does not equal to each other, [223] and [18]
2023-04-16 17:23:17 The column[0] value of row[28] does not equal to each other, [Supplier#000028778] and [Supplier#000030454]
2023-04-16 17:23:17 The column[1] value of row[28] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[29] does not equal to each other, [Supplier#000033129] and [Supplier#000038567]
2023-04-16 17:23:17 The column[1] value of row[29] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[30] does not equal to each other, [Supplier#000067630] and [Supplier#000045549]
2023-04-16 17:23:17 The column[1] value of row[30] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[31] does not equal to each other, [Supplier#000071795] and [Supplier#000056342]
2023-04-16 17:23:17 The column[1] value of row[31] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[32] does not equal to each other, [Supplier#000073047] and [Supplier#000057401]
2023-04-16 17:23:17 The column[1] value of row[32] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[33] does not equal to each other, [Supplier#000077118] and [Supplier#000058783]
2023-04-16 17:23:17 The column[1] value of row[33] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[34] does not equal to each other, [Supplier#000090078] and [Supplier#000067989]
2023-04-16 17:23:17 The column[1] value of row[34] does not equal to each other, [222] and [18]
2023-04-16 17:23:17 The column[0] value of row[35] does not equal to each other, [Supplier#000029402] and [Supplier#000070297]
2023-04-16 17:23:17 The column[1] value of row[35] does not equal to each other, [221] and [18]
2023-04-16 17:23:17 The column[0] value of row[36] does not equal to each other, [Supplier#000034927] and [Supplier#000073047]
2023-04-16 17:23:17 The column[1] value of row[36] does not equal to each other, [221] and [18]
2023-04-16 17:23:17 The column[0] value of row[37] does not equal to each other, [Supplier#000058653] and [Supplier#000076736]
2023-04-16 17:23:17 The column[1] value of row[37] does not equal to each other, [221] and [18]
2023-04-16 17:23:17 The column[0] value of row[38] does not equal to each other, [Supplier#000058916] and [Supplier#000082571]
2023-04-16 17:23:17 The column[1] value of row[38] does not equal to each other, [221] and [18]
2023-04-16 17:23:17 The column[0] value of row[39] does not equal to each other, [Supplier#000099850] and [Supplier#000085357]
2023-04-16 17:23:17 The column[1] value of row[39] does not equal to each other, [221] and [18]
2023-04-16 17:23:17 The column[0] value of row[40] does not equal to each other, [Supplier#000011461] and [Supplier#000087433]
2023-04-16 17:23:17 The column[1] value of row[40] does not equal to each other, [220] and [18]
2023-04-16 17:23:17 The column[0] value of row[41] does not equal to each other, [Supplier#000020610] and [Supplier#000090078]
2023-04-16 17:23:17 The column[1] value of row[41] does not equal to each other, [220] and [18]
2023-04-16 17:23:17 The column[0] value of row[42] does not equal to each other, [Supplier#000030025] and [Supplier#000091984]
2023-04-16 17:23:17 The column[1] value of row[42] does not equal to each other, [220] and [18]
2023-04-16 17:23:17 The column[0] value of row[43] does not equal to each other, [Supplier#000034034] and [Supplier#0000[96](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:97)390]
2023-04-16 17:23:17 The column[1] value of row[43] does not equal to each other, [220] and [18]
2023-04-16 17:23:17 The column[0] value of row[44] does not equal to each other, [Supplier#000038951] and [Supplier#0000[97](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:98)341]
2023-04-16 17:23:17 The column[1] value of row[44] does not equal to each other, [220] and [18]
2023-04-16 17:23:17 The column[0] value of row[45] does not equal to each other, [Supplier#000042108] and [Supplier#000000269]
2023-04-16 17:23:17 The column[1] value of row[45] does not equal to each other, [220] and [17]
2023-04-16 17:23:17 The column[0] value of row[46] does not equal to each other, [Supplier#000050265] and [Supplier#000001175]
2023-04-16 17:23:17 The column[1] value of row[46] does not equal to each other, [220] and [17]
2023-04-16 17:23:17 The column[0] value of row[47] does not equal to each other, [Supplier#000059617] and [Supplier#000002012]
2023-04-16 17:23:17 The column[1] value of row[47] does not equal to each other, [220] and [17]
2023-04-16 17:23:17 The column[0] value of row[48] does not equal to each other, [Supplier#000071107] and [Supplier#000003135]
2023-04-16 17:23:17 The column[1] value of row[48] does not equal to each other, [220] and [17]
2023-04-16 17:23:17 The column[0] value of row[49] does not equal to each other, [Supplier#000079503] and [Supplier#000005671]
2023-04-16 17:23:17 The column[1] value of row[49] does not equal to each other, [220] and [17]
2023-04-16 17:23:17 The column[0] value of row[50] does not equal to each other, [Supplier#000099520] and [Supplier#000011687]
2023-04-16 17:23:17 The column[1] value of row[50] does not equal to each other, [220] and [17]
2023-04-16 17:23:17 The column[0] value of row[51] does not equal to each other, [Supplier#000021117] and [Supplier#000013642]
2023-04-16 17:23:17 The column[1] value of row[51] does not equal to each other, [219] and [17]
2023-04-16 17:23:17 The column[0] value of row[52] does not equal to each other, [Supplier#000050948] and [Supplier#000013847]
2023-04-16 17:23:17 The column[1] value of row[52] does not equal to each other, [219] and [17]
2023-04-16 17:23:17 The column[0] value of row[53] does not equal to each other, [Supplier#000058783] and [Supplier#000014829]
2023-04-16 17:23:17 The column[1] value of row[53] does not equal to each other, [219] and [17]
2023-04-16 17:23:17 The column[0] value of row[54] does not equal to each other, [Supplier#000077886] and [Supplier#000016447]
2023-04-16 17:23:17 The column[1] value of row[54] does not equal to each other, [219] and [17]
2023-04-16 17:23:17 The column[0] value of row[55] does not equal to each other, [Supplier#0000784[98](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:99)] and [Supplier#000017373]
2023-04-16 17:23:17 The column[1] value of row[55] does not equal to each other, [219] and [17]
2023-04-16 17:23:17 The column[0] value of row[56] does not equal to each other, [Supplier#000000895] and [Supplier#00001[99](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:100)20]
2023-04-16 17:23:17 The column[1] value of row[56] does not equal to each other, [218] and [17]
2023-04-16 17:23:17 The column[0] value of row[57] does not equal to each other, [Supplier#000014526] and [Supplier#000021054]
2023-04-16 17:23:17 The column[1] value of row[57] does not equal to each other, [218] and [17]
2023-04-16 17:23:17 The column[0] value of row[58] does not equal to each other, [Supplier#000019210] and [Supplier#000021378]
2023-04-16 17:23:17 The column[1] value of row[58] does not equal to each other, [218] and [17]
2023-04-16 17:23:17 The column[0] value of row[59] does not equal to each other, [Supplier#000048113] and [Supplier#000023048]
2023-04-16 17:23:17 The column[1] value of row[59] does not equal to each other, [218] and [17]
2023-04-16 17:23:17 The column[0] value of row[60] does not equal to each other, [Supplier#000059612] and [Supplier#000024372]
2023-04-16 17:23:17 The column[1] value of row[60] does not equal to each other, [218] and [17]
2023-04-16 17:23:17 The column[0] value of row[61] does not equal to each other, [Supplier#000071242] and [Supplier#000027097]
2023-04-16 17:23:17 The column[1] value of row[61] does not equal to each other, [218] and [17]
2023-04-16 17:23:17 The column[0] value of row[62] does not equal to each other, [Supplier#000002567] and [Supplier#000030718]
2023-04-16 17:23:17 The column[1] value of row[62] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[63] does not equal to each other, [Supplier#000012673] and [Supplier#000031076]
2023-04-16 17:23:17 The column[1] value of row[63] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[64] does not equal to each other, [Supplier#000017799] and [Supplier#000034034]
2023-04-16 17:23:17 The column[1] value of row[64] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[65] does not equal to each other, [Supplier#000021732] and [Supplier#000034210]
2023-04-16 17:23:17 The column[1] value of row[65] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[66] does not equal to each other, [Supplier#000027097] and [Supplier#000036686]
2023-04-16 17:23:17 The column[1] value of row[66] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[67] does not equal to each other, [Supplier#000044362] and [Supplier#000037065]
2023-04-16 17:23:17 The column[1] value of row[67] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[68] does not equal to each other, [Supplier#000047474] and [Supplier#000038326]
2023-04-16 17:23:17 The column[1] value of row[68] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[69] does not equal to each other, [Supplier#000049647] and [Supplier#000040027]
2023-04-16 17:23:17 The column[1] value of row[69] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[70] does not equal to each other, [Supplier#000050896] and [Supplier#000044689]
2023-04-16 17:23:17 The column[1] value of row[70] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[71] does not equal to each other, [Supplier#000061395] and [Supplier#000045955]
2023-04-16 17:23:17 The column[1] value of row[71] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[72] does not equal to each other, [Supplier#000096404] and [Supplier#000046085]
2023-04-16 17:23:17 The column[1] value of row[72] does not equal to each other, [217] and [17]
2023-04-16 17:23:17 The column[0] value of row[73] does not equal to each other, [Supplier#000015700] and [Supplier#000046351]
2023-04-16 17:23:17 The column[1] value of row[73] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[74] does not equal to each other, [Supplier#000022490] and [Supplier#000047789]
2023-04-16 17:23:17 The column[1] value of row[74] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[75] does not equal to each other, [Supplier#000036545] and [Supplier#000051112]
2023-04-16 17:23:17 The column[1] value of row[75] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[76] does not equal to each other, [Supplier#000046896] and [Supplier#000051985]
2023-04-16 17:23:17 The column[1] value of row[76] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[77] does not equal to each other, [Supplier#000047592] and [Supplier#000052695]
2023-04-16 17:23:17 The column[1] value of row[77] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[78] does not equal to each other, [Supplier#000050658] and [Supplier#000054151]
2023-04-16 17:23:17 The column[1] value of row[78] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[79] does not equal to each other, [Supplier#000053942] and [Supplier#000056706]
2023-04-16 17:23:17 The column[1] value of row[79] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[80] does not equal to each other, [Supplier#000060218] and [Supplier#000060692]
2023-04-16 17:23:17 The column[1] value of row[80] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[81] does not equal to each other, [Supplier#000068000] and [Supplier#000064139]
2023-04-16 17:23:17 The column[1] value of row[81] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[82] does not equal to each other, [Supplier#000091469] and [Supplier#000065018]
2023-04-16 17:23:17 The column[1] value of row[82] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[83] does not equal to each other, [Supplier#000097189] and [Supplier#000065198]
2023-04-16 17:23:17 The column[1] value of row[83] does not equal to each other, [216] and [17]
2023-04-16 17:23:17 The column[0] value of row[84] does not equal to each other, [Supplier#000007074] and [Supplier#000067040]
2023-04-16 17:23:17 The column[1] value of row[84] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[85] does not equal to each other, [Supplier#000014829] and [Supplier#000068049]
2023-04-16 17:23:17 The column[1] value of row[85] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[86] does not equal to each other, [Supplier#000023048] and [Supplier#000068083]
2023-04-16 17:23:17 The column[1] value of row[86] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[87] does not equal to each other, [Supplier#000024695] and [Supplier#000072296]
2023-04-16 17:23:17 The column[1] value of row[87] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[88] does not equal to each other, [Supplier#000027383] and [Supplier#000074107]
2023-04-16 17:23:17 The column[1] value of row[88] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[89] does not equal to each other, [Supplier#000041978] and [Supplier#000077698]
2023-04-16 17:23:17 The column[1] value of row[89] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[90] does not equal to each other, [Supplier#000048048] and [Supplier#000085875]
2023-04-16 17:23:17 The column[1] value of row[90] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[91] does not equal to each other, [Supplier#000053519] and [Supplier#000088029]
2023-04-16 17:23:17 The column[1] value of row[91] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[92] does not equal to each other, [Supplier#000078385] and [Supplier#000088880]
2023-04-16 17:23:17 The column[1] value of row[92] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[93] does not equal to each other, [Supplier#000081764] and [Supplier#000094467]
2023-04-16 17:23:17 The column[1] value of row[93] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[94] does not equal to each other, [Supplier#000094467] and [Supplier#000098259]
2023-04-16 17:23:17 The column[1] value of row[94] does not equal to each other, [215] and [17]
2023-04-16 17:23:17 The column[0] value of row[95] does not equal to each other, [Supplier#000002389] and [Supplier#000099288]
2023-04-16 17:23:17 The column[1] value of row[95] does not equal to each other, [214] and [17]
2023-04-16 17:23:17 The column[0] value of row[96] does not equal to each other, [Supplier#000005145] and [Supplier#000000207]
2023-04-16 17:23:17 The column[1] value of row[96] does not equal to each other, [214] and [16]
2023-04-16 17:23:17 The column[0] value of row[97] does not equal to each other, [Supplier#000012685] and [Supplier#000000722]
2023-04-16 17:23:17 The column[1] value of row[97] does not equal to each other, [214] and [16]
2023-04-16 17:23:17 The column[0] value of row[98] does not equal to each other, [Supplier#000012891] and [Supplier#000000895]
2023-04-16 17:23:17 The column[1] value of row[98] does not equal to each other, [214] and [16]
2023-04-16 17:23:17 The column[0] value of row[99] does not equal to each other, [Supplier#000022904] and [Supplier#000002744]
2023-04-16 17:23:17 The column[1] value of row[99] does not equal to each other, [214] and [16]
2023-04-16 17:23:17 The column[0] value of row[[100](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:101)] does not equal to each other, [Supplier#000027696] and [Supplier#000005145]
2023-04-16 17:23:17 The column[1] value of row[100] does not equal to each other, [214] and [16]
2023-04-16 17:23:17 The query q21  has failed to  be executed,and cost: 11545

2023-04-16 17:23:17 Now start to execute the query queries/q22.sql,please wait.....

2023-04-16 17:23:18 The query_id is: 60ab7ac8-dc7b-11ed-8bad-fe133a2b8513
2023-04-16 17:23:18 Now start to compare query result for [report/TPCH_10/q22.txt,golden/TPCH_10/q22.txt]
2023-04-16 17:23:18 The column[1] value of row[1] does not equal to each other, [36164] and [9053]
2023-04-16 17:23:18 The column[2] value of row[1] does not equal to each other, [27[101](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:102)9520.25] and [67752399.56]
2023-04-16 17:23:18 The column[1] value of row[2] does not equal to each other, [36044] and [9006]
2023-04-16 17:23:18 The column[2] value of row[2] does not equal to each other, [27[108](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:109)2301.92] and [67894525.19]
2023-04-16 17:23:18 The column[1] value of row[3] does not equal to each other, [36347] and [9078]
2023-04-16 17:23:18 The column[2] value of row[3] does not equal to each other, [273152761.73] and [68177380.15]
2023-04-16 17:23:18 The column[1] value of row[4] does not equal to each other, [36703] and [9240]
2023-04-16 17:23:18 The column[2] value of row[4] does not equal to each other, [275422017.09] and [69425181.57]
2023-04-16 17:23:18 The column[1] value of row[5] does not equal to each other, [36223] and [8990]
2023-04-16 17:23:18 The column[2] value of row[5] does not equal to each other, [271528796.52] and [67271840.24]
2023-04-16 17:23:18 The column[1] value of row[6] does not equal to each other, [36395] and [9013]
2023-04-16 17:23:18 The column[2] value of row[6] does not equal to each other, [272941991.42] and [67705001.84]
2023-04-16 17:23:18 The column[1] value of row[7] does not equal to each other, [35995] and [9006]
2023-04-16 17:23:18 The column[2] value of row[7] does not equal to each other, [270[114](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:115)040.42] and [67600913.56]
[202](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/4714177429/jobs/8360854873#step:15:203)3-04-16 17:23:18 The query q22  has failed to  be executed,and cost: 979

[tpch_q21_hung_cn0.txt](https://github.com/matrixorigin/matrixone/files/11245509/tpch_q21_hung_cn0.txt)
[tpch_q21_hung_cn1.txt](https://github.com/matrixorigin/matrixone/files/11245511/tpch_q21_hung_cn1.txt)

### Steps to Reproduce

```Markdown
1. run a mo cluster on eks with (2*cn, 1*dn, 3*log)
2. load tpch 10g data 
3. run tpch 10g q21 query

or 

1. git clone https://github.com/matrixorigin/mo-auto-test.git
2. modify .github/workflows/regression-on-distributed-mode.yml, 
  set JobName.default to tpch_benchmark_test
  set RunTimes to 1
3. make a pr to merge, and the job will automatically run the tpch 10g queries on eks
```

**Comment by aunjgr** [comment]
I found the reason now. The current RIGHT join implementation is also wrong, but not exposed by tpch.

Still working on a feasible solution.

**Comment by aunjgr** [comment]
Current `Dispatch` operator doesn't have shuffle mode. Both RIGHT OUTER join and RIGHT ANTI join yield wrong result on multi CN, because we don't summarize the `matched` flags across CNs.

It's better force RIGHT joins to run on single CN right now. When shuffle mode is ready, we will revisit these joins.

**Comment by aunjgr** [comment]
The workaround PR has been merged.

@aressu1985 Please verify.

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
fixed commit:1084951
